docs/source/data/how_to_create_a_custom_dataset.md

```diff
@@ -1,7 +1,12 @@
-# Custom datasets
+# Advanced: Tutorial to create a custom dataset
 
 [Kedro supports many datasets](/kedro_datasets) out of the box, but you may find that you need to create a custom dataset. For example, you may need to handle a proprietary data format or filesystem in your pipeline, or perhaps you have found a particular use case for a dataset that Kedro does not support. This tutorial explains how to create a custom dataset to read and save image data.
 
+## AbstractDataSet
+
+For contributors, if you would like to submit a new dataset, you must extend the [`AbstractDataSet` interface](/kedro.io.AbstractDataSet), which is the underlying interface that all datasets extend. It requires subclasses to override the `_load` and `_save` and provides `load` and `save` methods that enrich the corresponding private methods with uniform error handling. It also requires subclasses to override `_describe`, which is used in logging the internal information about the instances of your custom `AbstractDataSet` implementation.
```

Comment:
I think we should push to implement `AbstractVersionedDataSet` rather than `AbstractDataSet` in most of the case.

Reply:
Slight problem with this is that the example goes on to use `PartitionedDataSet` so it doesn't support versioning IIRC. I can reword to say that you should use the versioned class as the parent but there will be cases where you can't, and this example doesn't. Then we can make a ticket to separately update the custom dataset text more completely (as I think it's beyond the scope of my changes right now).

Reply:
New ticket https://github.com/kedro-org/kedro/issues/2937